conans/client/tools/net.py

```diff
@@ -60,21 +86,38 @@ def download(url, filename, verify=True, out=None, retry=None, retry_wait=None,
     retry_wait = retry_wait if retry_wait is not None else config.retry_wait
     retry_wait = retry_wait if retry_wait is not None else 5
 
-    downloader = FileDownloader(requester=requester, output=out, verify=verify, config=config)
+    url = [url] if not isinstance(url, (list, tuple)) else url
     checksum = sha256 or sha1 or md5
-    # The download cache is only used if a checksum is provided, otherwise, a normal download
+
+    downloader = FileDownloader(requester=requester, output=out, verify=verify, config=config)
     if config and config.download_cache and checksum:
         downloader = CachedFileDownloader(config.download_cache, downloader, user_download=True)
-        downloader.download(url, filename, retry=retry, retry_wait=retry_wait, overwrite=overwrite,
-                            auth=auth, headers=headers, md5=md5, sha1=sha1, sha256=sha256)
-    else:
-        downloader.download(url, filename, retry=retry, retry_wait=retry_wait, overwrite=overwrite,
-                            auth=auth, headers=headers)
-        if md5:
-            check_md5(filename, md5)
-        if sha1:
-            check_sha1(filename, sha1)
-        if sha256:
-            check_sha256(filename, sha256)
 
-    out.writeln("")
+    def _download_file(downloader, url):
+        # The download cache is only used if a checksum is provided, otherwise, a normal download
+        if isinstance(downloader, CachedFileDownloader):
+            downloader.download(url, filename, retry=retry, retry_wait=retry_wait,
+                                overwrite=overwrite, auth=auth, headers=headers, md5=md5,
+                                sha1=sha1, sha256=sha256)
+        else:
+            downloader.download(url, filename, retry=retry, retry_wait=retry_wait,
+                                overwrite=overwrite, auth=auth, headers=headers)
+            if md5:
+                check_md5(filename, md5)
+            if sha1:
+                check_sha1(filename, sha1)
+            if sha256:
+                check_sha256(filename, sha256)
+        out.writeln("")
+
+    for index, url_it in enumerate(url):
+        try:
+            _download_file(downloader, url_it)
+            break
+        except ConanException as error:
+            message = "Could not download from the URL {}: {}.".format(url_it, str(error))
+            if len(url) == 1:
+                raise ConanException(message)
+            out.error(message)
```

Comment:
One question about this line: will someone parse this `ERROR` as an actual error and think that the command has failed? Is it an error to fail with an URL when you have a mirror/fallback strategy? Printing something to `stderr` (we are not printing errors to `stderr`, should we?) is considered a failure from a CLI perspective?

wdyt, @memsharded @uilianries @czoido ?

Reply:
This is a very good point, indeed.
Probably should be a warning, not an error. Not a failure from CLI perspective, the command have succeeded if it was able to fallback to a mirror.

Reply:
Okay, I'll update it to warning level.